src/worker.cpp

```diff
@@ -143,37 +75,72 @@ void *Worker::jpeg_grabber(void *arg)
     LOG_DEBUG("Start jpeg_grabber thread.");
 
     StartHelper *sh = static_cast<StartHelper *>(arg);
-    int encChn = sh->encChn;
+    int jpgChn = sh->encChn - 2;
     int ret;
-    struct timeval ts;
-    gettimeofday(&ts, NULL);
+    uint32_t bps{0};
+    int delay{0};
+    uint32_t fps{(uint32_t)global_jpeg[jpgChn]->stream->fps};
+    int64_t auto_fps{(int64_t)((1000000/global_jpeg[jpgChn]->stream->fps) * 0.8)};
 
-    global_jpeg->imp_encoder = IMPEncoder::createNew(global_jpeg->stream, encChn, global_jpeg->stream->jpeg_channel, "stream2");
+    unsigned long long ms{0};
+    gettimeofday(&global_jpeg[jpgChn]->stream->stats.ts, NULL);
+    global_jpeg[jpgChn]->stream->stats.ts.tv_sec -= 10;
+
+    global_jpeg[jpgChn]->imp_encoder = IMPEncoder::createNew(
+        global_jpeg[jpgChn]->stream, sh->encChn, global_jpeg[jpgChn]->stream->jpeg_channel, "stream2");
 
     // inform main that initialization is complete
     sh->has_started.release();
 
-    ret = IMP_Encoder_StartRecvPic(global_jpeg->encChn);
-    LOG_DEBUG_OR_ERROR(ret, "IMP_Encoder_StartRecvPic(" << global_jpeg->encChn << ")");
+    ret = IMP_Encoder_StartRecvPic(global_jpeg[jpgChn]->encChn);
```

Comment:
We have `0` hard coded sometimes but here we use `jpgChn`. Should we fix the inconsistencies?

Reply:
in preparation to have more than 1 JPEG channel in future.